crates/sui-verifier/src/verifier.rs

```diff
@@ -3,16 +3,28 @@
 
 //! This module contains the public APIs supported by the bytecode verifier.
 
-use move_binary_format::file_format::CompiledModule;
-use sui_types::error::ExecutionError;
+use move_binary_format::{access::ModuleAccess, file_format::CompiledModule};
+use move_core_types::identifier::IdentStr;
+use sui_types::{error::ExecutionError, SUI_FRAMEWORK_ADDRESS};
 
 use crate::{
     entry_points_verifier, global_storage_access_verifier, id_leak_verifier,
     one_time_witness_verifier, private_generics, struct_with_key_verifier,
 };
 
+const TEST_SCENARIO_MODULE_NAME: &str = "test_scenario";
+
 /// Helper for a "canonical" verification of a module.
 pub fn verify_module(module: &CompiledModule) -> Result<(), ExecutionError> {
+    if *module.address() == SUI_FRAMEWORK_ADDRESS
+        && module.name() == IdentStr::new(TEST_SCENARIO_MODULE_NAME).unwrap()
```

Comment:
Which of these cases do we break? It's probably just private generics, right?

Reply:
If so, can we add the exclusion just to that verifier?

Reply:
> Which of these cases do we break? It's probably just private generics, right?

This was the first one, but I suspected that `test_scenario` may have more.

> If so, can we add the exclusion just to that verifier?

Are you thinking excluding altogether or just for `test_scenario`? If it's the latter, then it will be a bit hacky and special-case-ish but certainly doable - instead of excluding `test_scenario` at the outermost verifier level, I can pass info on whether a specific module should be excluded deeper to individual passes.

I was not sure if it's worth it for `test_scenario`, to be honest, though, but I defer to you whether I should pursue it or not.

Reply:
I think it might be worthwhile to limit precisely where test_scenario is bypassing things, since a lot of these checks are there to prevent the system from blowing up. While it's just tests, it might be nice to have that extra layer of protection. And I think just private generics is the only spot where this is an issue

Reply:
Sounds good! I was not sure what I kind of can of worms I might be opening here but indeed it seems like private generics for transfers are the only issue here.